metaflow/plugins/kfp/tests/run_integration_tests.py

```diff
@@ -100,19 +103,60 @@ def test_s3_sensor_flow(pytestconfig) -> None:
     return
 
 
-# this test ensures the integration tests fail correctly
-def test_raise_failure_flow(pytestconfig) -> None:
+# This test ensures the integration tests fail correctly
+# and that when a test fails, an OpsGenie email is sent.
+# It verifies the OpsGenie email has been sent by querying
+# the OpsGenie API to look for the corresponding alert
+# and then closes that alert.
+def test_error_and_opsgenie_alert(pytestconfig) -> None:
     test_cmd = (
         f"{_python()} flows/raise_error_flow.py --datastore=s3 kfp run "
         f"--wait-for-completion --workflow-timeout 1800 "
-        f"--experiment metaflow_test --tag test_t1 "
+        f"--experiment metaflow_test --tag test_t1 --notify "
     )
     if pytestconfig.getoption("image"):
         test_cmd += (
             f"--no-s3-code-package --base-image {pytestconfig.getoption('image')}"
         )
 
-    exponential_backoff_from_platform_errors(test_cmd, 1)
+    kfp_run_id = exponential_backoff_from_platform_errors(test_cmd, 1)
+    # TODO use @secret (when it's released) to store this API key
```

Comment:
This is an interesting one, I think we've got some thinking about how we handle importing secrets given the current strategy DevEx has for storing and linking secrets. I'd actually say to remove the TODO as I think this approach is fine and maybe if/when we get to `@secret` if it makes sense to add this in.

Reply:
Cool, removed the TODO. I forgot about this TODO and I don't think it's needed anymore.